be/src/runtime/client_cache.cpp

```diff
@@ -73,26 +69,19 @@ Status ClientCacheHelper::get_client(const TNetworkAddress& hostport, ClientFact
 Status ClientCacheHelper::reopen_client(ClientFactory& factory_method, void** client_key,
                                         int timeout_ms) {
     DCHECK(*client_key != nullptr) << "Trying to reopen nullptr client";
-    ThriftClientImpl* client_to_close = nullptr;
+    std::shared_ptr<ThriftClientImpl> client_to_close;
```

Comment:
Add macro  ENABLE_FACTORY_CREATOR to ThriftClientImpl in order to force using smart pointer please.